src/Storage/Device/S3.php

```diff
@@ -686,6 +686,12 @@ private function call(string $method, string $uri, string $data = '', array $par
         }
 
         $this->headers['date'] = \gmdate('D, d M Y H:i:s T');
+        
+        // if cn zone selected, switch to amazonaws.cn
+        if ($this->region == 'cn-north-4' || $this->region == 'cn-northwest-1') {
```

Comment:
Would you please use the constants instead of hard-coded magic strings?

Reply:
done 👍